lib/ansible/modules/database/mysql/mysql_user.py

```diff
@@ -427,14 +427,14 @@ def pick(x):
             return x
 
     for grant in grants:
-        res = re.match("GRANT (.+) ON (.+) TO '.*'@'.*'( IDENTIFIED BY PASSWORD '.+')? ?(.*)", grant[0])
+        res = re.match("""GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\5)? ?(.*)""", grant[0])
```

Comment:
Just a note: regexes should really be compiled outside of loops and then use the compiled regex's methods inside of the loop.  ie:
``` python
grant_re = re.compile("""GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\5)? ?(.*)""")
for grant in grants:
    res = grant_re.match(grant[0])
```

regexes also are also usually deserving of a comment explaining what they do, what cornercases they attempt to handle, and some sample inputs which illustrate those to make it easier to review and modify the regex in the future.

Also, I believe there is a bug in the new regex.  To match the closing quote, you have ```\\3``` and ```\\4``` but you only have ```\5```.  I believe that should be ```\\5```.

Reply:
If that is a bug, first fix it in devel and then cherry-pick the additional commit to this PR so that the correct code can be merged as one unit to the stable-2.7 branch.